smaug/python/ops/control_flow_ops.py

```diff
@@ -30,9 +30,77 @@ def merge(input_tensors, name="merge"):
   Returns:
     A tensor that the available input tensor forwards to.
   """
-  return add_node(
-      name=name,
-      op=Merge,
-      input_tensors=input_tensors,
+  return common.add_node(
+      name=name, op=types_pb2.Merge, input_tensors=input_tensors,
       output_tensors_dims=[input_tensors[0].shape.dims],
       output_tensor_layout=input_tensors[0].shape.layout)[0]
+
+def cond(predication, true_fn, false_fn, name="cond"):
+  """A conditional operator.
+
+  This operator provides the capability of doing if-else statement. Depending on
+  the predication value, either the True or the False body of the operator will
+  be executed.
+
+  Args:
+    predication: A predication tensor of value 0 or 1, determining which path to
+      execute.
+    true_fn: The callable to be performed if `predication` is 1.
+    false_fn: The callable to be performed if `predication` is 0.
+
+  Returns:
+    The tensors returned by either true_fn or false_fn.
+  """
+
+  def _insert_switch_nodes(predication, branch, graph):
+    """Insert switch nodes for external tensros in the subgraph."""
```

Comment:
* typo in tensros.
* `branch` is unclear. You use it to index the output of the `switch` node that you added, but that's not a great abstraction. I would change this to `branch_result` and figure out what the index is internally (or make it part of the SwitchOp API).
* please explain in the comment what an "external" tensor is. I think really what you mean is "all input tensors to either branch of the conditional op".

In summary, a doc comment explaining what this function is doing and the argument types is useful (this is Python after all).

Reply:
- Fixed typo.
- As suggested, changed `branch` to `branch_result` and internally determined which result of the switch node to use.
- Added API comment.